File: Example/ObjectiveC Example/Channel Plugin Sample objc/LoginViewController.m
//
//  LoginViewController.m
//  Channel Plugin Sample objc
//
//  Created by Haeun Chung on 21/03/2017.
//  Copyright © 2017 ZOYI. All rights reserved.
//

#import "LoginViewController.h"
#import "MainViewController.h"

@interface LoginViewController () <UITextFieldDelegate>

@end

@implementation LoginViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
  
  self.idField.placeholder = @"user Id";
  self.usernameField.placeholder = @"user name (optional)";
  self.phoneField.placeholder = @"mobile number (optional)";
  self.warningLabel.text = @"";
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)didClickOnLogin {
  [self.idField resignFirstResponder];
  [self.phoneField resignFirstResponder];
  [self.usernameField resignFirstResponder];
  
  if (![self.idField.text isEqualToString:@""]) {
    self.warningLabel.text = @"";
    [self performSegueWithIdentifier:@"MainViewSegue" sender:nil];
  } else {
    self.warningLabel.text = @"id is required to login as user";
  }
}
  
- (BOOL)textFieldShouldReturn:(UITextField *)textField {
  [self.idField resignFirstResponder];
  [self.phoneField resignFirstResponder];
  [self.usernameField resignFirstResponder];
  return YES;
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
  if ([[segue identifier] isEqualToString:@"MainViewSegue"]) {
    MainViewController *viewController = [segue destinationViewController];
    viewController.isUser = YES;
    viewController.userId = self.idField.text;
  }
}

@end

File: Example/Swift Example/Channel Plugin Sample/LoginViewController.swift
//
//  LoginViewController.swift
//  Example
//
//  Created by Haeun Chung on 14/03/2017.
//  Copyright © 2017 ZOYI. All rights reserved.
//

import Foundation
import UIKit

class LoginViewController : UIViewController {
  @IBOutlet var idField: UITextField!
  @IBOutlet var usernameField: UITextField!
  @IBOutlet var phoneField: UITextField!
  @IBOutlet var loginButton: UIButton!
  @IBOutlet var warningLabel: UILabel!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    self.idField.placeholder = "user Id"
    self.usernameField.placeholder = "user name (optional)"
    self.phoneField.placeholder = "mobile number (optional)"
    self.warningLabel.text = ""
  }
  
  override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
    if let destinationVC = segue.destination as? MainViewController {
      destinationVC.isUser = true
      destinationVC.userId = self.idField.text ?? ""
      destinationVC.userName = self.usernameField.text ?? ""
      destinationVC.phoneNumber = self.phoneField.text ?? ""
    }
  }
  
  @IBAction func didClickOnLogin() {
    self.idField.resignFirstResponder()
    self.usernameField.resignFirstResponder()
    self.phoneField.resignFirstResponder()
    
    if let id = self.idField.text {
      self.warningLabel.text = ""
      self.performSegue(withIdentifier: "MainViewSegue", sender: id)
    } else {
      self.warningLabel.text = "id is required to login as user"
    }
  }
}

extension LoginViewController: UITextFieldDelegate {
  func textFieldShouldReturn(_ textField: UITextField) -> Bool {
    self.idField.resignFirstResponder()
    self.usernameField.resignFirstResponder()
    self.phoneField.resignFirstResponder()
    return true
  }
}
